Backend/src/config/passport-setup.ts

```diff
@@ -0,0 +1,61 @@
+import passport from 'passport';
+import { Strategy as GoogleStrategy } from 'passport-google-oauth20';
+import { type users } from '@prisma/client'; // Make sure to import the User type from Prisma
+import keys from './keys';
+import UserService from '@/modules/users/users.service';
+
+passport.serializeUser((user: users, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser(async (id: string, done) => {
+  try {
+    const userService = new UserService();
+    const user = await userService.findUserById(id);
+
+    if (user) {
+      done(null, user);
+    } else {
+      done(new Error('User not found'), null);
+    }
+  } catch (error) {
+    console.error('Error deserializing user:', error);
+    done(error, null);
+  }
+});
+
+export default passport.use(
+  new GoogleStrategy(
+    {
+      // options for google strategy
+      clientID: keys.google.clientID,
+      clientSecret: keys.google.clientSecret,
+      callbackURL:
+        'http://localhost:3000/api/v1/development/users/auth/google/redirect',
+    },
+    async (accessToken, refreshToken, profile, done) => {
+      // passport callback function
+      console.log('passport callback function fired:');
+      const name: string = profile.displayName;

```

Comment:
We need to get user profile picture too.

Reply:
ok, I will add it.